rfc/0057-sdk3-analytics.md

```diff
@@ -191,6 +191,27 @@ class AnalyticsWarning {
 }
 ```
 
+## Cluster compatibility
+With the upcoming release of Enterprise Analytics, and the risk of users accidentally using operational SDKs against it, a compatibility check is being added.
+
+[MB-67103](https://jira.issues.couchbase.com/browse/MB-67103) adds to the cluster and bucket configs a `prodName` string field identifying the cluster type.
+
+Before performing each analytics operation, check the cluster config, and iff the prodName field is present, check if it starts with "Couchbase Server".  If it doesn't, fast-fail the request with a generic `CouchbaseException`.  A suitable error message could be:
```

Comment:
We ended up doing something slightly different to this. We are checking that the "prod" (not "prodName") field does not have the value "analytics". See https://jira.issues.couchbase.com/browse/CBD-6390